platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/map/MapTest.java

```diff
@@ -104,4 +123,51 @@ void insertRemoveAndModifyOneMillion() throws InterruptedException {
             map.release();
         }
     }
+
+    @Test
+    @Tags({@Tag("VirtualMerkle")})
+    @DisplayName("Delete a value that was moved to a different virtual path")
+    void deletedObjectLeavesOnFlush() throws InterruptedException {
+        VirtualMap<TestObjectKey, TestValue> map = createObjectMap("deletedObjectLeavesOnFlush");
+        for (int i = 0; i < 8; i++) {
+            map.put(new TestObjectKey(i), new TestValue(i));
+        }
+
+        VirtualRootNode<TestObjectKey, TestValue> rootNode = map.getRight();
+        rootNode.enableFlush();
+
+        RecordAccessor<TestObjectKey, TestValue> records = rootNode.getRecords();
+        // Check that key/value 0 is at path 7
+        VirtualLeafRecord<TestObjectKey, TestValue> leaf = records.findLeafRecord(7, false);
+        assertNotNull(leaf);
+        assertEquals(new TestObjectKey(0), leaf.getKey());
+        assertEquals(new TestValue(0), leaf.getValue());
+
+        VirtualMap<TestObjectKey, TestValue> copy = map.copy();
+        map.release();
+        map = copy;
+        rootNode.waitUntilFlushed();
+
+        // Move key/value to a different path, then delete
+        map.remove(new TestObjectKey(0));
+        map.remove(new TestObjectKey(2));
+        map.put(new TestObjectKey(8), new TestValue(8));
+        map.put(new TestObjectKey(0), new TestValue(0));
+        map.remove(new TestObjectKey(0));
+
+        rootNode = map.getRight();
+        rootNode.enableFlush();
+
+        copy = map.copy();
+        map.release();
+        map = copy;
+        rootNode.waitUntilFlushed();
+
+        // During this second flush, key/value 0 must be deleted from the map despite it's
+        // path the virtual tree doesn't match the path in the data source
```

Comment:
Nit: this might need rephrasing: "...despite it's path the virtual..."

Reply:
Looks like "in" is missed...